analysis.go

```diff
@@ -128,6 +148,31 @@ func getDumpedJSONResourceFactory(basepath []string) DumpedJSONResourceFactory {
 
 }
 
+// CheckForWaitingPods checks all pods for any containers in waiting status
+func CheckForWaitingPods(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
+	result := Result{Status: 0, StatusMessage: "this issue was not detected", CheckName: "check pods for 'waiting' containers", Info: []Info{}, Events: []Event{}}
+	pods := Pods{}
+	if err := JSONResourceFactory([]string{"definitions", "pods.json"}, &pods); err != nil {
+		result.Status = 2
+		result.StatusMessage = "Error executing task"
+		return result, err
+	}
+
+	for _, pod := range pods.Items {
```

Comment:
pretty sure that the slice will be initialised even if  there are no items , but will pod.Status.ContainerStatuses be initialised / have a value or could you hit a nil pointer here?